Userland/Libraries/LibJS/Runtime/Value.cpp

```diff
@@ -486,23 +486,8 @@ ThrowCompletionOr<Value> Value::to_number(VM& vm) const
     case BOOLEAN_TAG:
         return Value(as_bool() ? 1 : 0);
     case STRING_TAG: {
-        String string = Utf8View(as_string().string()).trim(whitespace_characters, AK::TrimMode::Both).as_string();
-        if (string.is_empty())
-            return Value(0);
-        if (string == "Infinity" || string == "+Infinity")
-            return js_infinity();
-        if (string == "-Infinity")
-            return js_negative_infinity();
-        char* endptr;
-        auto parsed_double = strtod(string.characters(), &endptr);
-        if (*endptr)
-            return js_nan();
-        // NOTE: Per the spec only exactly [+-]Infinity should result in infinity
-        //       but strtod gives infinity for any case-insensitive 'infinity' or 'inf' string.
-        if (isinf(parsed_double) && string.contains('i', AK::CaseSensitivity::CaseInsensitive))
-            return js_nan();
-
-        return Value(parsed_double);
+        auto primitive = TRY(to_primitive(vm, PreferredType::Number));
+        return primitive.string_to_number();
```

Comment:
Since we know the Value is a string here we don't need to do to_primitive. In fact this could be visible so we're kind of not allowed to.
We can just return `string_to_number()`.

Reply:
Now that you mention it I really don't know why I chose to do that. Thanks!